server/controllers/events/events_controller.go

```diff
@@ -681,6 +686,16 @@ func (e *VCSEventsController) HandleAzureDevopsPullRequestEvent(w http.ResponseW
 		}
 	}
 
+	resource, ok := event.Resource.(*azuredevops.GitPullRequest)
```

Comment:
L699 `e.Parser.ParseAzureDevopsPullEvent(*event)` do the same things. Could you consider followings?

- use `e.Parser.ParseAzureDevopsPullEvent(*event)` result to get repository url
- if prText is fixed on test event, add `ignoreEvents`.

Reply:
I'll have a think about this. 
In terms of the parser, it could be used... It does point to a differently formatted URL than the one I'm using (though I think it might work)
But I would need to revert b30913b9acfae4bd96293df353a221e506fde724 and set it back to looking at strings because the objects are different types which might make it a bit messier than just leaving it alone. 

For ignore events, it is possible to use that method but the event text is `Jamal Hartnett created a new pull request` which might have an unfortunate side effect on anyone named Jamal Hartnett

Reply:
how about to use `baseRepo`? Both cases can get baseRepo after parsing by EventParser.

```go
// azure devops test webhook always contains https://fabrikam.visualstudio.com/DefaultCollection/_apis/git/repositories/4bc14d40-c903-45e2-872e-0462c7748079.
const azureDevOpsTestRepoVCSHost = "fabrikam.visualstudio.com"
const azureDevOpsTestRepoFullName = "DefaultCollection/_apis/..." // <- This value is parsed by repository.WebURL, not URL. I'm not sure the correct value.
func isAzureDevOpsTestRepoURL(baseRepo models.Repo) bool {
	return baseRepo.VCSHost.Hostname == azureDevOpsTestRepoVCSHost && baseRepo.FullName == azureDevOpsTestRepoFullName
}
```

By the way `https://fabrikam.visualstudio.com/DefaultCollection/_apis/git/repositories/4bc14d40-c903-45e2-872e-0462c7748079` looks like a magic number for me. If there is some docs on azuread or something, let me know and please add comment like `// azure devops test webhook always contains https://fabrikam.visualstudio.com/DefaultCollection/_apis/git/repositories/4bc14d40-c903-45e2-872e-0462c7748079`.

Reply:
If you can get a collect webhook value, you can test EventParser behaviour by https://github.com/runatlantis/atlantis/blob/9345d55127b822c0617818a369919ea24e1762b7/server/events/event_parser_test.go#L1203-L1219.

Reply:
I've spent a while bashing my head against this one and I think I would like to propose to leave it as is. 

The issue that I've found is that the test webhook json (the body can be found in the PR description) is not the same format as real requests. I was trying to get values from `e.Parser.ParseAzureDevopsPullEvent(*event)` but it panics every time because the `webUrl` property doesn't exist! 
Because of this I'm not able to use the `baseRepo` object as it's unable to complete the parsing function. I might add some comments in the code to this effect.

Thoughts?

Reply:
@SSKLCP  I see.

Could you extract test url with `const` to explicit it as azure devops const value not to use it as magic number.

Reply:
Done now 🙂